// warframe-worldstate-parser/dist/DuviriCycle-CmHJStJo.js
import { WorldstateObject } from "./WorldstateObject-BSTYuZwm.js";

//#region lib/models/DuviriCycle.ts
const cycleTime = 36e3;
const stateTime = 7200;
const states = [
	"sorrow",
	"fear",
	"joy",
	"anger",
	"envy"
];
const getStageInfo = () => {
	const cycleDelta = (Math.floor(Date.now() / 1e3) - 52) % cycleTime;
	const stateInd = Math.floor(cycleDelta / stateTime);
	const stateDelta = cycleDelta % stateTime;
	const untilNext = stateTime - stateDelta;
	const expiry = new Date(Date.now() + untilNext * 1e3);
	expiry.setSeconds(0);
	expiry.setMilliseconds(0);
	const activation = /* @__PURE__ */ new Date(expiry.getTime() - stateTime * 1e3);
	activation.setSeconds(0);
	activation.setMilliseconds(0);
	return {
		state: states[stateInd],
		expiry,
		activation
	};
};
/**
* @typedef {WorldstateObject} DuviriCycle
* @augments {WorldstateObject}
*/
var DuviriCycle = class extends WorldstateObject {
	/**
	* Current stage key
	*/
	state;
	/**
	* Choice options for this Cycle
	*/
	choices;
	/**
	* @param duviriChoices The current circuit choices
	*/
	constructor(duviriChoices) {
		super({ _id: { $oid: "duviriCycle0" } });
		const { activation, expiry, state } = getStageInfo();
		this.activation = activation;
		this.expiry = expiry;
		this.state = state;
		this.choices = duviriChoices;
		this.id = `duviriCycle${this.state}${this.expiry.getTime()}`;
	}
};

//#endregion
export { DuviriCycle };
